mycroft/skills/common_query_skill.py

```diff
@@ -80,39 +105,76 @@ def __handle_question_query(self, message):
             level = result[1]
             answer = result[2]
             callback = result[3] if len(result) > 3 else None
-            confidence = self.__calc_confidence(match, search_phrase, level)
-            self.bus.emit(message.response({"phrase": search_phrase,
-                                            "skill_id": self.skill_id,
-                                            "answer": answer,
-                                            "callback_data": callback,
-                                            "conf": confidence}))
+            confidence = self.__calc_confidence(match, search_phrase, level, answer)
+            self.bus.emit(
+                message.response(
+                    {
+                        "phrase": search_phrase,
+                        "skill_id": self.skill_id,
+                        "answer": answer,
+                        "callback_data": callback,
+                        "conf": confidence,
+                    }
+                )
+            )
         else:
             # Signal we are done (can't handle it)
-            self.bus.emit(message.response({"phrase": search_phrase,
-                                            "skill_id": self.skill_id,
-                                            "searching": False}))
-
-    def __calc_confidence(self, match, phrase, level):
+            self.bus.emit(
+                message.response(
+                    {
+                        "phrase": search_phrase,
+                        "skill_id": self.skill_id,
+                        "searching": False,
+                    }
+                )
+            )
+
+    def remove_noise(self, phrase):
+        # remove noise to produce essence
+        phrase = " " + phrase + " "
+        for word in self.translated_noise_words:
+            mtch = " " + word + " "
+            if phrase.find(mtch) > -1:
+                phrase = phrase.replace(word, "")
```

Comment:
Will there be an issue here if there is for example an "and" and the word "andy"? Example "he is called andy and he is cool"

I have previously made good use of `re.sub()` and the "word boundry" `\b`in cases like this.

This function could be:
```python
        for word in self.translated_noise_words:
            phrase = re.sub(r'\b' + word + r'\b', '', phrase)
        return " ".join(phrase.split())
```

Or possibly lever the regex logic even more
```python
        # match any whole word that exists in the list of noise words
        regex = r'\b(' + '|'.join(self.translated_noise_words) + r')\b'
        result = re.sub(regex, '', phrase)
        return ' '.join(result.split())
```

Reply:
the method i suggest porting over from OVOS about should account for this and for finding the voc files

```python
    def remove_voc(self, utt, voc_filename, lang=None):
        """ removes any entry in .voc file from the utterance """
        lang = lang or self.lang
        cache_key = lang + voc_filename

        if cache_key not in self.voc_match_cache:
			# this call avoids duplicate logic to load .voc
            self.voc_match(utt, voc_filename, lang)

        if utt:
            # Check for matches against complete words
            for i in self.voc_match_cache.get(cache_key) or []:
                # Substitute only whole words matching the token
                utt = re.sub(r'\b' + i + r"\b", "", utt)

        return utt
 ```

Reply:
Yeah it uses that very regex :)

Reply:
Addressed in last commit